packages/design-system/src/components/datalist/datalist.js

```diff
@@ -239,6 +239,7 @@ const Datalist = forwardRef(function Datalist(
         selectButtonStylesOverride={highlightStylesOverride || focusStyle}
         aria-label={dropdownButtonLabel}
         isOpen={isOpen}
+        disabled={disabled}
```

Comment:
We may have to check other places in the codebase. It is a good fix.

Reply:
I hope TS will help catch this kind of stuff